ecosystem/typescript/sdk_v2/src/crypto/ed25519.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { Deserializer, Serializer } from "../bcs";
+import { Hex } from "../core";
+import { HexInput } from "../types";
+
+export class Ed25519PublicKey {
+  // Correct length of the public key in bytes (Uint8Array)
+  static readonly LENGTH: number = 32;
+
+  // The public key in hex format
+  readonly data: Hex;
+
+  constructor(hexInput: HexInput) {
```

Comment:
let's make sure we have the same syntax in all classes
```suggestion
  constructor(args:{data: HexInput}) {
```